[[ Bug 19658 ]] Make CEF work in Linux

runrevmark opened the pull request:
This patch changes the layout of the CEF files so that things will
run from the -bin folder when building from source, from the IDE
and in Standalones.

Note: This is still not quite right - but getting there - there is a problem with the placement of the various files. Basically the *-cefprocess files, icudat and *.bin files must be next to the LiveCode executable; whilst libcef and other resources can be in a subfolder. However, when running from the bin folder, the subfolder is CEF, and not Externals/CEF.

Depends on: https://github.com/livecode/livecode-thirdparty/pull/103

runrevmark committed:
[[ Bug 19658 ]] Make CEF work in Linux

This patch changes the layout of the CEF files so that things will
run from the -bin folder when building from source, from the IDE
and in Standalones.

runrevmark committed:
[[ Bug 19658 ]] Make CEF work on Linux

This patch further changes the layout of CEF components on Linux.

The -bin folder generated from building on Linux now reflects the
same structure as required at runtime:

   engine
   *-cefprocess
   icu.dat
   *.bin
   Externals/CEF/libcef.so
   Externals/CEF/*

This is also reflected in the installer layout, both for 'Runtime'
components, and the set of externals for the IDE as well as what
is generated by the standalone builder.

livecodeali commented:
@livecode-vulcan review ok 765da37

livecode-vulcan commented:
💙 review by @livecodeali ok 765da370488e2bce5961cfee0cc4284489eedcc2

livecode-vulcan commented:
😎 test success 765da370488e2bce5961cfee0cc4284489eedcc2

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/369)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/375)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/355)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/298)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/298)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/364)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/369)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/367)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/373)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/367)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/373)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/392)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/363)

runrevmark committed:
[[ Bug 19658 ]] Tweak resources path for Linux

This patch ensures the CEF resources are fetched from the
correct place in the SB, whether it be installed or BFS
install.
